xla/debug_options_flags.cc

```diff
@@ -982,42 +1016,14 @@ void MakeDebugOptionsFlags(std::vector<tsl::Flag>* flag_list,
                 debug_options->xla_gpu_deterministic_ops(),
                 "Guarantees run-to-run determinism on GPU."));
   flag_list->push_back(tsl::Flag(
-      "xla_gpu_enable_async_collectives",
-      bool_setter_for(&DebugOptions::set_xla_gpu_enable_async_collectives),
-      debug_options->xla_gpu_enable_async_collectives(),
-      "Converts synchronous collective ops into asynchronous."));
-  flag_list->push_back(tsl::Flag(
-      "xla_gpu_enable_async_all_reduce",
-      bool_setter_for(&DebugOptions::set_xla_gpu_enable_async_all_reduce),
-      debug_options->xla_gpu_enable_async_all_reduce(),
-      "Converts synchronous all-reduce ops into asynchronous."));
-  flag_list->push_back(tsl::Flag(
-      "xla_gpu_enable_async_collective_broadcast",
-      bool_setter_for(
-          &DebugOptions::set_xla_gpu_enable_async_collective_broadcast),
-      debug_options->xla_gpu_enable_async_collective_broadcast(),
-      "Converts synchronous collective-broadcast ops into asynchronous."));
-  flag_list->push_back(tsl::Flag(
-      "xla_gpu_enable_async_collective_permute",
-      bool_setter_for(
-          &DebugOptions::set_xla_gpu_enable_async_collective_permute),
-      debug_options->xla_gpu_enable_async_collective_permute(),
-      "Converts synchronous collective-permute ops into asynchronous."));
-  flag_list->push_back(tsl::Flag(
-      "xla_gpu_enable_async_all_gather",
-      bool_setter_for(&DebugOptions::set_xla_gpu_enable_async_all_gather),
-      debug_options->xla_gpu_enable_async_all_gather(),
-      "Converts synchronous all-gather ops into asynchronous."));
-  flag_list->push_back(tsl::Flag(
-      "xla_gpu_enable_async_reduce_scatter",
-      bool_setter_for(&DebugOptions::set_xla_gpu_enable_async_reduce_scatter),
-      debug_options->xla_gpu_enable_async_reduce_scatter(),
-      "Converts synchronous reduce-scatter ops into asynchronous."));
-  flag_list->push_back(tsl::Flag(
-      "xla_gpu_enable_async_all_to_all",
-      bool_setter_for(&DebugOptions::set_xla_gpu_enable_async_all_to_all),
-      debug_options->xla_gpu_enable_async_all_to_all(),
-      "Converts synchronous all-to-all ops into asynchronous."));
+      "xla_gpu_disable_async_collectives",
+      setter_for_xla_gpu_disable_async_collectives,
+      collective_op_types_to_string(
+          debug_options->xla_gpu_disable_async_collectives()),
+      "This disables a certain set of async collectives and turn them into"
+      " synchornous ones. By default, this is empty which indicates enabling"
+      " async execution for all collectives. A sample usage is: "
+      " --xla_gpu_disable_async_collectives=ALLREDUCE,REDUCESCATTER"));
```

Comment:
@Tixxx 
How to disable all collectives?
It would be good to have a list of all collective somewhere, or a special value like "ALL" to disable all of them.